storage/src/test/java/org/apache/kafka/storage/internals/log/RemoteIndexCacheTest.java

```diff
@@ -0,0 +1,1207 @@
+            // and if cache is non-existed, the cache file should not exist.
+            if (getIndexFileFromRemoteCacheDir(cache, LogFileUtils.INDEX_FILE_SUFFIX).isPresent()) {
+                assertCacheSize(1);
+            } else {
+                assertCacheSize(0);
+            }
+        } finally {
+            executor.shutdownNow();
+        }
+    }
+
+    @Test
+    public void testMultipleIndexEntriesExecutionInCorruptException() throws IOException, RemoteStorageException {
+        reset(rsm);
+        when(rsm.fetchIndex(any(RemoteLogSegmentMetadata.class), any(IndexType.class))).thenAnswer(ans -> {
+            RemoteLogSegmentMetadata metadata = ans.getArgument(0);
+            IndexType indexType = ans.getArgument(1);
+            OffsetIndex offsetIdx = createOffsetIndexForSegmentMetadata(metadata, tpDir);
+            TimeIndex timeIdx = createTimeIndexForSegmentMetadata(metadata, tpDir);
+            TransactionIndex txnIdx = createTxIndexForSegmentMetadata(metadata, tpDir);
+            maybeAppendIndexEntries(offsetIdx, timeIdx);
+            // Create corrupted index file
+            createCorruptTimeIndexOffsetFile(tpDir);
+            return switch (indexType) {
+                case OFFSET -> new FileInputStream(offsetIdx.file());
+                case TIMESTAMP -> new FileInputStream(timeIdx.file());
+                case TRANSACTION -> new FileInputStream(txnIdx.file());
+                case LEADER_EPOCH -> null; // leader-epoch-cache is not accessed.
+                case PRODUCER_SNAPSHOT -> null; // producer-snapshot is not accessed.
+            };
+        });
+
+        assertThrows(CorruptIndexException.class, () -> cache.getIndexEntry(rlsMetadata));
+        assertNull(cache.internalCache().getIfPresent(rlsMetadata.remoteLogSegmentId().id()));
+        verifyFetchIndexInvocation(1, List.of(IndexType.OFFSET, IndexType.TIMESTAMP));
+        verifyFetchIndexInvocation(0, List.of(IndexType.TRANSACTION));
+        // Current status
+        // (cache is null)
+        // RemoteCacheDir contain
+        // 1. Offset Index File is fine and not corrupted
+        // 2. Time Index File is corrupted
+        // What should be the code flow in next execution
+        // 1. No rsm call for fetching OffSet Index File.
+        // 2. Time index file should be fetched from remote storage again as it is corrupted in the first execution.
+        // 3. Transaction index file should be fetched from remote storage.
+        reset(rsm);
+        // delete all files created in tpDir
+        List<Path> paths = Files.walk(tpDir.toPath(), 1)
+                .filter(Files::isRegularFile)
+                .toList();
+        for (Path path : paths) {
+            Files.deleteIfExists(path);
+        }
+        // rsm should return no corrupted file in the 2nd execution
+        when(rsm.fetchIndex(any(RemoteLogSegmentMetadata.class), any(IndexType.class))).thenAnswer(ans -> {
```

Comment:
this is totally equal to `mockRsmFetchIndex`, right?

Reply:
Good catch! It is indeed identical